File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.java
class Solution {
    public String reverseWords(String s) {
        String[] arr = s.split(" ");
        StringBuilder reversed = new StringBuilder();

        for (int j = 0; j < arr.length; j++) {
            String word = arr[j];
            char[] chars = word.toCharArray();

            for (int i = 0; i < word.length() / 2; i++) {
                char temp = chars[i];
                chars[i] = chars[word.length() - 1 - i];
                chars[word.length() - 1 - i] = temp;
            }

            reversed.append(new String(chars));
            if (j < arr.length - 1) {
                reversed.append(" ");
            }
        }

        return reversed.toString();
    }
}

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    // Split the input string into an array of words
    let words = s.split(" ");

    // Reverse each word in the array
    let reversedWords = words.map(word => {
        return word.split("").reverse().join("");
    });

    // Join the reversed words back into a single string
    let reversedSentence = reversedWords.join(" ");

    return reversedSentence;

};